src/binary_operator.cpp

```diff
@@ -225,6 +225,38 @@ double eval_dbl(const binary_operator &bo, const std::unordered_map<std::string,
     }
 }
 
+long double eval_ldbl(const binary_operator &bo, const std::unordered_map<std::string, long double> &map,
```

Comment:
Would be better to abstract this in a separate implementation template in which you use ``eval<T>(...)``.